File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.github.pagehelper.Page;
import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

import java.util.List;

/**
 * 品牌接口
 * @author fiveqiao
 */
public interface BrandService {
    List<TbBrand> findAll();

    /**
     * 品牌分页
     * @param pageNum 当前页面
     * @param pageSize  每页记录数
     * @return
     */
    PageResult<TbBrand> findPage(TbBrand brand,int pageNum,int pageSize);
//    Page<TbBrand> findPage(int pageNum, int pageSize);

    /**
     * 添加 brand
     * @param brand
     */
    void add(TbBrand brand);

    /**
     * 根据id查询实体
     * @param id
     * @return
     */
    TbBrand findOne(long id);

    /**
     *修改brand
     * @param brand
     */
    void update(TbBrand brand);

    /**
     * 删除
     * @param ids
     */
    void delete(Long[] ids);

}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
//品牌服务
app.service("brandService", function ($http) {
    //查询所有
    this.findAll = function () {
        return $http.get("../brand/findAll.do");
    };
    //分页查询
    this.findPage = function (page, size) {
        return $http.get("../brand/findPage.do?page=" + page + "&size=" + size);
    };
    //查询单个
    this.findOne = function (id) {
        return $http.get("../brand/findOne.do?id=" + id);
    };
    //增加
    this.add = function (entity) {
        return $http.post("../brand/add.do", entity);
    };
    //修改
    this.update = function (entity) {
        return $http.post("../brand/update.do", entity);
    };
    //删除
    this.dele = function (ids) {
        return $http.get("../brand/delete.do?ids=" + ids);
    };
    //搜索
    this.search = function (page, size, searchEntity) {
        return $http.post("../brand/search.do?page=" + page + "&size=" + size, searchEntity);
    };
    //下拉列表数据
    this.selectOptionList= function () {
        return $http.get("../brand/selectOptionList.do");
    };
})